ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/PendingTransactions.java

```diff
@@ -85,6 +125,11 @@ public void removeTransaction(final Transaction transaction) {
                     transactionsBySender.remove(transaction.getSender());
                   }
                 });
+        if (removedTransactionInfo.isReceivedFromLocalSource()) {
+          localTransactionCounter.inc(-1);
+        } else {
+          remoteTransactionCounter.inc(-1);
```

Comment:
Counters should never be decremented, they only go up.  We should have a separate "transactions_dropped_total" counter that we increment here (probably with a "source" label for local & remote).